dev_guide/builds.adoc

```diff
@@ -1184,6 +1190,80 @@ spec:
 ----
 ====
 
+[[build-hooks]]
+
+== Build Hooks
+Build hooks can be used as a part of the build flow to run tests inside a temporary container and validate your image before it is being pushed to the registry. The command is executed with the current working directory set to the image's `WORKDIR`.
+The hook will fail if the script or command returns a non-zero exit code or if there's any other error related to starting the temporary container. When the hook fails it will mark the build as failed.
+====
```

Comment:
This section between `====` is good to describe what the fields in `postCommit` are, but it does so with an invalid example. I'd suggest we rework this content to be with the numbered examples below (1, 2, ..., 5).

That is:

```
... When the hook fails it will mark the build as failed.

There are five different ways to configure the post build hook....

(include details within the examples below)
1. ...
2. ...
...
```